src/card/interface/submitCardFields.js

```diff
@@ -32,70 +87,23 @@ export function submitCardFields({
     featureFlags,
   });
 
+  // $FlowIssue
+  const [isPurchaseFlow, isVaultFlow] = [Boolean(cardProps.createOrder), Boolean(cardProps.createVaultSetupToken)];
```

Comment:
```suggestion
  const [isPurchaseFlow, isWithoutPurchaseFlow] = [Boolean(cardProps.createOrder), Boolean(cardProps.createVaultSetupToken)];
```

Nit: I think the `isVaultFlow` naming could be a little misleading - the `isPurchaseFlow` supports a one time checkout and also a vault with purchase. One idea is `isWithoutPurchase` curious on thoughts from others cc: @bywood @wsbrunson @wickerpopstar

Reply:
yeah, specifically `isVaultWithoutPurchase` because there are also other non-purchase flows